source/docs/hardware-reference/talonfx/improving-performance-with-current-limits.rst

```diff
@@ -0,0 +1,217 @@
+
+.. math::
+
+   V_{loaded} = V_{unloaded} - (I_{totalsupply} * R_{battery})
+
+.. raw:: html
+
+   <h4>Loaded Battery Voltage Calculator</h4>
+   <div style="width:100%; overflow:hidden;">
+      <form onkeypress="return event.keyCode != 13" style="float:left;">
+         <p>Unloaded voltage (V)</p>
+         <input onchange="updateOutput()" id="uV" value="12.5" style="width:90%;" type="numeric" placeholder="12.5"/>
+      </form>
+      <form onkeypress="return event.keyCode != 13" style="float:left;">
+         <p>Total current draw (A)</p>
+         <input onchange="updateOutput()" id="current" value="240" style="width:90%;" type="numeric" placeholder="240"/>
+      </form>
+      <form onkeypress="return event.keyCode != 13" style="float:left;">
+         <p>Battery resistance (mOhms)</p>
+         <input onchange="updateOutput()" id="resistance" value="20" style="width:90%;" type="numeric" placeholder="20"/>
+      </form>
+      <p style="float:left;margin-left:10px;margin-top:35px;font-weight:bold;color:#bdeb34;">= <span id="output">0 V</span></p>
+   </div>
+   <br/>
+
+   <script>
+      updateOutput();
+
+      function updateOutput() {
+         var unloadedVoltage = document.getElementById("uV").value
+         var current = document.getElementById("current").value
+         var resistance = document.getElementById("resistance").value
+         var output = document.getElementById("output")
+
+         var calculatedOutput = parseFloat(unloadedVoltage) - (parseFloat(current) * (parseFloat(resistance) / 1000))
+
+         output.innerHTML = (Math.round(calculatedOutput*10**2)/10**2) + " V"
+      }
+   </script>
+
+Be aware that battery health (in the form of battery resistance above) significantly impacts how large current draw affects the output voltage of the battery. Health of the battery can be roughly determined via a `battery beak <https://store.ctr-electronics.com/battery-beak/>`__ or by performing a battery discharge test with a `battery analyzer <https://www.andymark.com/products/computerized-battery-analyzer>`__.
+
+Using the above information, ensure your battery is healthy and that your current limits will prevent the battery from sagging below 7 V.
+
+Preventing Wheel Slip
+^^^^^^^^^^^^^^^^^^^^^
+
+Stator current limits are excellent at preventing wheel slip (thus increasing traction). To determine wheel slip, perform the following instructions.
+
+1. Place the robot on carpet against a wall.
+2. Begin plotting velocity and stator current in :doc:`Tuner X </docs/tuner/plotting>`.
+3. Slowly increase voltage output until the velocity becomes non-zero.
+
+Set your stator current limit to a value below the observed stator current in Tuner. In the example below, the wheels began slipping at around 130 A.
+
+.. image:: images/slip-current.png
+   :alt: Wheel slip at 130 A stator current
+
+Limiting Acceleration
+^^^^^^^^^^^^^^^^^^^^^
```

Comment:
Do we want a section on limiting acceleration? We already have one for wheel slip, and to me limiting acceleration is more a side effect than an intended use case; the primary goal is limiting torque moreso than acceleration.

I'm not sure if there's a way we can do it, but I'd like to somehow document using stator current limits to prevent mechanical damage (i.e. running into hard stops).

I think it would also be valuable to have a section on using stator current limits to prevent heat buildup in high-inertia systems such as a flywheel. I can gather data for some plots.